frontend/packages/dev-console/src/components/add/AddPage.tsx

```diff
@@ -19,8 +27,73 @@ export const PageContents: React.FC<AddPageProps> = ({ match }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
 
+  const [warningMessageFlag, setWarningMessageFlag] = React.useState<boolean>();
+  const [resourceQuotaName, setResourceQuotaName] = React.useState<string>('');
+  const [resourceQuotaKind, setResourceQuotaKind] = React.useState<string>('');
+
+  const [quotas, rqLoaded] = useK8sWatchResource<ResourceQuotaKind[]>({
+    groupVersionKind: {
+      kind: ResourceQuotaModel.kind,
+      version: ResourceQuotaModel.apiVersion,
+    },
+    namespace,
+    isList: true,
+  });
+
+  const [clusterQuotas, acrqLoaded] = useK8sWatchResource<AppliedClusterResourceQuotaKind[]>({
+    groupVersionKind: {
+      kind: AppliedClusterResourceQuotaModel.kind,
+      version: AppliedClusterResourceQuotaModel.apiVersion,
+      group: AppliedClusterResourceQuotaModel.apiGroup,
+    },
+    namespace,
+    isList: true,
+  });
+
+  const [totalRQatQuota, quotaName, quotaKind] = checkQuotaLimit(quotas);
+  const [totalACRQatQuota, clusterRQName, clusterRQKind] = checkQuotaLimit(clusterQuotas);
+
+  let totalResourcesAtQuota = [...totalRQatQuota, ...totalACRQatQuota];
+  totalResourcesAtQuota = totalResourcesAtQuota.filter((resourceAtQuota) => resourceAtQuota !== 0);
+
+  React.useEffect(() => {
+    if (totalResourcesAtQuota.length === 1) {
+      setResourceQuotaName(quotaName || clusterRQName);
+      setResourceQuotaKind(quotaKind || clusterRQKind);
+    } else {
+      setResourceQuotaName('');
+      setResourceQuotaKind('');
+    }
+    if (totalResourcesAtQuota.length > 0) {
+      setWarningMessageFlag(true);
+    } else {
+      setWarningMessageFlag(false);
+    }
+  }, [clusterRQKind, clusterRQName, totalResourcesAtQuota, quotaKind, quotaName]);
+
+  const getRedirectLink = () => {
+    if (resourceQuotaName && resourceQuotaKind === AppliedClusterResourceQuotaModel.kind) {
+      return `/k8s/ns/${namespace}/${AppliedClusterResourceQuotaModel.apiGroup}~${AppliedClusterResourceQuotaModel.apiVersion}~${AppliedClusterResourceQuotaModel.kind}/${resourceQuotaName}`;
+    }
+    if (resourceQuotaName) {
+      return `/k8s/ns/${namespace}/${ResourceQuotaModel.plural}/${resourceQuotaName}`;
+    }
+    return `/k8s/ns/${namespace}/${ResourceQuotaModel.plural}`;
+  };
+
   return namespace ? (
-    <AddPageLayout title={t('devconsole~Add')} />
+    <>
+      {warningMessageFlag && rqLoaded && acrqLoaded ? (
+        <Alert variant="warning" title={t('devconsole~Resource quota reached')} isInline>
+          <Link to={getRedirectLink()}>
+            {t('devconsole~{{count}} resource reached quota', {
+              count: totalResourcesAtQuota.reduce((a, b) => a + b, 0),
+            })}
+          </Link>
+        </Alert>
+      ) : null}
+      <AddPageLayout title={t('devconsole~Add')} />
+    </>
```

Comment:
@lokanandaprabhu I will suggest that you extract the code from this file and create a separate ResourceQuotaBanner component and use that in Add/Topology pages so that we can avoid duplicating code.